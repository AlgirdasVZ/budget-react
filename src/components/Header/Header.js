import React from 'react';
import { StyledNavLink } from './Header.style';
import { startLogout } from '../../actions/auth';
import { connect } from 'react-redux';

export const Header = ({ startLogout }) => (
  <header>
    <h1>Expensify</h1>

    <StyledNavLink to="/dashboard" activeClassName="is-active">Dashboard</StyledNavLink>
    <StyledNavLink to="/create" activeClassName="is-active">Add Expense</StyledNavLink>
    <StyledNavLink to="/help" activeClassName="is-active">Help</StyledNavLink>
    <button onClick={startLogout}>Logout</button>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);