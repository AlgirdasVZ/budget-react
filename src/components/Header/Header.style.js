import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

export const StyledNavLink = styled(NavLink)`
  &.is-active {
    font-weight: bold;
  }
`;