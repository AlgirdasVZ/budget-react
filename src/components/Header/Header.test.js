import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

test('should render Header correctly', () => {
  const wrapper = shallow(<Header />);

  expect(wrapper).toMatchSnapshot();
});
