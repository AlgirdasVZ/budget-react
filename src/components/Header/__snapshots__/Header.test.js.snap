// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <Styled(NavLink)
    activeClassName="is-active"
    to="/dashboard"
  >
    Dashboard
  </Styled(NavLink)>
  <Styled(NavLink)
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </Styled(NavLink)>
  <Styled(NavLink)
    activeClassName="is-active"
    to="/help"
  >
    Help
  </Styled(NavLink)>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
